src/diracx/db/jobs/status_utility.py

```diff
@@ -146,3 +150,195 @@ async def set_job_status(
         )
 
     return SetJobStatusReturn(**job_data)
+
+
+async def delete_job(
+    job_id: int,
+    config: Config,
+    job_db: JobDB,
+    job_logging_db: JobLoggingDB,
+    task_queue_db: TaskQueueDB,
+    background_task: BackgroundTasks,
+):
+    """
+    Delete a job by killing and setting the job status to DELETED.
+    """
+
+    await __kill_delete_job(
+        job_id,
+        config,
+        job_db,
+        task_queue_db,
+        background_task,
+    )
+
+    await set_job_status(
+        job_id,
+        {
+            datetime.now(timezone.utc): JobStatusUpdate(
+                Status=JobStatus.DELETED,
+                MinorStatus="Checking accounting",
+                StatusSource="job_manager",
+            )
+        },
+        job_db,
+        job_logging_db,
+        force=True,
+    )
+
+
+async def kill_job(
+    job_id: int,
+    config: Config,
+    job_db: JobDB,
+    job_logging_db: JobLoggingDB,
+    task_queue_db: TaskQueueDB,
+    background_task: BackgroundTasks,
+):
+    await __kill_delete_job(
+        job_id,
+        config,
+        job_db,
+        task_queue_db,
+        background_task,
+    )
+
+    await set_job_status(
+        job_id,
+        {
+            datetime.now(timezone.utc): JobStatusUpdate(
+                Status=JobStatus.KILLED,
+                MinorStatus="Marked for termination",
+                StatusSource="job_manager",
+            )
+        },
+        job_db,
+        job_logging_db,
+        force=True,
+    )
+
+
+async def __kill_delete_job(
+    job_id: int,
+    config: Config,
+    job_db: JobDB,
+    task_queue_db: TaskQueueDB,
+    background_task: BackgroundTasks,
+):
+    from DIRAC.Core.Utilities.ReturnValues import returnValueOrRaise
+    from DIRAC.StorageManagementSystem.Client.StorageManagerClient import (
+        StorageManagerClient,
+    )
+
+    res = await job_db.search(
+        parameters=["Status", "Owner", "OwnerGroup"],
+        search=[
+            ScalarSearchSpec(
+                parameter="JobID",
+                operator=ScalarSearchOperator.EQUAL,
+                value=job_id,
+            ),
+        ],
+        sorts=[],
+    )
+    if not res:
+        raise NoResultFound(f"Job {job_id} not found")
+
+    status = res[0]["Status"]
+    owner = res[0]["Owner"]
+    owner_group = res[0]["OwnerGroup"]
+    vo = await job_db.get_vo(job_id)
+
+    if status == JobStatus.STAGING:
+        returnValueOrRaise(StorageManagerClient().killTasksBySourceTaskID([job_id]))
```

Comment:
We can't use the client here. At the very least it needs to be a `raise NotImplementedError` and a `TODO`.

Reply:
I just chatted with @chaen and he thinks we should just be a TODO and not even bother with the `NotImplementedError`.